base/generator.jl

```diff
@@ -0,0 +1,21 @@
+"""
+    Generator(f, iter)
+
+Given a function `f` and an iterator `iter`, construct an iterator that yields
+the values of `f` applied to the elements of `iter`.
+The syntax `f(x) for x in iter` is syntax for constructing an instance of this
+type.
+"""
```

Comment:
Actually I wonder about the name of this type. The syntax for it is definitely called a "generator expression" (at least in python), but in python this produces a generator object, which is somewhat similar to a Task, while we return more of a MapIterator. I guess best not to export it for now.